numpy/random/mtrand.pyx

```diff
@@ -4149,12 +4153,14 @@ cdef class RandomState:
                [3, 4, 5]])
 
         """
-        if isinstance(x, (int, np.integer)):
+        arr = np.asarray(x)
+        if arr.ndim < 1 and arr.dtype.kind == 'i':
```

Comment:
If we guard the old API well, I guess we should not change it here in either case?